### PR TITLE
Bug 1593374 - Migrate mozaggregator to python 3

### PR DESCRIPTION
**Pull request by acmiyaguchi**
[bug link](https://bugzilla.mozilla.org/show_bug.cgi?id=1593374)

I've tested that the job still runs on Spark with python 3: https://dbc-caf9527b-e073.cloud.databricks.com/#job/4935/run/1

```
./mozetl-databricks.py \
	--git-path https://github.com/mozilla/python_mozaggregator.git \
	--git-branch python3 \
	--python 3 \
	--num-workers 5 \
	--token $DB_TOKEN \
	--module-name mozaggregator \
	mobile \
	--output s3://telemetry-test-bucket/mozaggregator/mobile_test/v2/ \
	--num-partitions 200
```

This will require a coordinate PR with telemetry-airflow to set the default python version to 3. I've also tested that the service can come online by running `make up`.

### CODE REVIEW

File: mozaggregator/cli.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def run_parquet(date, channels, output, num_partitions, source, project_id, data
 def run_mobile(date, output, num_partitions, source, project_id, dataset_id):
     spark = SparkSession.builder.getOrCreate()
 
-    print("Running job for {}".format(date))
+    print(("Running job for {}".format(date)))
```
Comment (robhudson) [inline mozaggregator/cli.py]: Are these extra nested parens needed? Or was that an over-aggressive search/replace?

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM python:2-stretch
+FROM python:3.7-stretch
```
Comment (robhudson) [inline Dockerfile]: Python 3.8 is out now.

File: tests/test_aggregator.py [hunk 3]
```diff
@@ -88,20 +88,20 @@ def test_simple_measurements(build_id_aggregates):
                 assert(value["histogram"][str(d.SIMPLE_SCALAR_BUCKET)] == value["count"])
 
     assert len(metric_count) == len(d.simple_measurements_template)
-    for process_counts in metric_count.values():
+    for process_counts in list(metric_count.values()):
```
Comment (robhudson) [inline tests/test_aggregator.py]: `values()` returns an iterable so no need to convert to a list if not needed. Same for other areas of the code.

File: tests/dataset.py [hunk 4]
```diff
@@ -189,72 +189,72 @@ def generate_pings():
...
     }
 
     child_payloads = [{"simpleMeasurements": simple_measurements_template}
                       for i in range(NUM_CHILDREN_PER_PING)]
 
-    scalars = dict(chain(scalars_template.iteritems(), ignored_scalars_template.iteritems()))
-    keyed_scalars = dict(chain(keyed_scalars_template.iteritems(),
-                               ignored_keyed_scalars_template.iteritems(),
-                               private_keyed_scalars_template.iteritems()))
+    scalars = dict(chain(iter(scalars_template.items()), iter(ignored_scalars_template.items())))
```
Comment (fbertsch) [inline tests/dataset.py]: Not sure if the `iter` is necessary here.

**Pull request merged**